Sofa/Component/Mapping/NonLinear/src/sofa/component/mapping/nonlinear/AreaMapping.inl

```diff
@@ -0,0 +1,376 @@
+using sofa::type::dot;
+
+template <class TIn, class TOut>
+AreaMapping<TIn, TOut>::AreaMapping()
+    : l_topology(initLink("topology", "link to the topology container"))
+{}
+
+template <class TIn, class TOut>
+auto AreaMapping<TIn, TOut>::computeSecondDerivativeArea(
+    const sofa::type::fixed_array<sofa::type::Vec3, 3>& triangleVertices) ->
+    sofa::type::Mat<3, 3, sofa::type::Mat<3, 3, Real> >
+{
+    const auto& v = triangleVertices;
+
+    const auto N = sofa::type::cross(v[1] - v[0], v[2] - v[0]);
+    const auto n2 = dot(N, N);
+
+    sofa::type::MatNoInit<3, 3, sofa::type::MatNoInit<3, 3, Real>> d2A;
+
+    const auto ka = 1 / (2 * std::sqrt(std::pow(n2, 3)));
+
+    static constexpr auto skewSign = type::crossProductMatrix(sofa::type::Vec<3, Real>{1,1,1});
+
+    for (unsigned int i = 0; i < 3; ++i)
+    {
+        for (unsigned int j = 0; j < 3; ++j)
+        {
+            auto& entry = d2A[i][j];
+
+            const auto i1 = (i + 1) % 3;
+            const auto j1 = (j + 1) % 3;
+            const auto i2 = (i + 2) % 3;
+            const auto j2 = (j + 2) % 3;
+
+            const auto N_cross_Pi1Pi2 = N.cross(v[i1] - v[i2]);
+            const auto N_cross_Pj1Pj2 = N.cross(v[j1] - v[j2]);
+
+            const auto outer_a = dyad(N_cross_Pi1Pi2, N_cross_Pj1Pj2);
+            static const auto& id = sofa::type::Mat<3, 3, SReal>::Identity();
+
+            const auto dot_product = dot(v[i1] - v[i2], v[j1] - v[j2]);
+            const auto outer_b = dyad(v[j1] - v[j2], v[i1] - v[i2]);
+
+            entry = - outer_a + n2 * (dot_product * id - outer_b);
+
+            if (i != j) // diagonal blocks are skipped because skewSign[i][j] == 0
+            {
+                const auto sign = skewSign[i][j];
+                entry += sign * n2 * type::crossProductMatrix(N);
+            }
+
+            entry *= ka;
+        }
+    }
+
+    return d2A;
+}
+
+template <class TIn, class TOut>
+void AreaMapping<TIn, TOut>::init()
+{
+    if (l_topology.empty())
+    {
+        msg_warning() << "link to Topology container should be set to ensure right behavior. First Topology found in current context will be used.";
+        l_topology.set(this->getContext()->getMeshTopologyLink());
+    }
+
+    msg_info() << "Topology path used: '" << l_topology.getLinkedPath() << "'";
+
+    const auto nbSurfacePrimitives = l_topology->getNbQuads() + l_topology->getNbTriangles();
+
+    if (nbSurfacePrimitives == 0)
+    {
+        msg_error() << "No topology component containing surface primitives found at path: " << l_topology.getLinkedPath() << ", nor in current context: " << this->getContext()->name;
+        this->d_componentState.setValue(sofa::core::objectmodel::ComponentState::Invalid);
+        return;
+    }
+
+    typename core::behavior::MechanicalState<In>::ReadVecCoord pos = this->getFromModel()->readPositions();
+
+    this->getToModel()->resize( nbSurfacePrimitives );
+    jacobian.resizeBlocks(nbSurfacePrimitives, pos.size());
```

Comment:
Isn't this useless ? It is then cleared in `apply`. Or does it perform smth like a `reserve` ?